pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -62,6 +78,64 @@ def visit_call(self, node):
         if isinstance(node.parent, (astroid.For, astroid.Comprehension)):
             self.add_message("consider-iterating-dictionary", node=node)
 
+    def _check_consider_using_str_partition(self, node: astroid.Call) -> None:
+        """Add message when accessing first or last elements of a str.split() or str.rsplit(),
+        or when split/rsplit with max_split=1 is used"""
+
+        # Check if call is split() or rsplit()
+        if isinstance(node.func, astroid.Attribute) and node.func.attrname in (
+            "split",
+            "rsplit",
+        ):
+            inferred_func = utils.safe_infer(node.func)
+            fn_name = node.func.attrname
+            node_name = node.as_string()
+
+            if not isinstance(inferred_func, astroid.BoundMethod):
+                return
+            try:
+                seperator = utils.get_argument_from_call(node, 0, "sep").value
+            except utils.NoSuchArgumentError:
+                return
+
+            # Check if maxsplit is set, and ignore checking if maxsplit is > 1
+            try:
+                maxsplit = utils.get_argument_from_call(node, 1, "maxsplit").value
+                if maxsplit > 1:
+                    return
+
+            except utils.NoSuchArgumentError:
+                maxsplit = 0
+
+            # Check if it's immediately subscripted
+            if isinstance(node.parent, astroid.Subscript):
+                subscript_node = node.parent
+                # Check if subscripted with -1/0
+                try:
+                    subscript_value = utils.get_subscript_const_value(
+                        subscript_node
+                    ).value
+                except ValueError:
+                    return
+
+                if maxsplit == 1 and subscript_value == 1:
+                    new_func = node.func.attrname.replace("split", "partition")
+                    new_name = f"{node.as_string().rpartition('.')[0]}.{new_func}({seperator})[-1]"
+                    self.add_message(
+                        "consider-using-str-partition", node=node, args=(new_name,)
+                    )
```

Comment:
You can move the `self.add_message` below both `if` statements if you change the second one to `elif` and add a `else` clause that returns.

Reply:
resolved: 23e82eb